### PR TITLE
Added NeoForge Support

### PR DESCRIPTION
**Pull request by MichaelHillcox**
Hey, thought you might find this helpful. I ported this for my own use case but seeing as it was relatively simple, I thought I might as well send over a PR.

I've tried to keep the project as like yours as possible, so I've kept the gradle setup... Even though this forced me to do a kinda ugly hack around the mods.toml file changes between Forge and Neoforge.

Please let me know if you'd like anything changed.

It's worth noting, remapping for your use case would likely be fine but in the future, the two versions will likely need larger changes.

PS: I have tested this in production and it seems to behave :D

### CODE REVIEW

**Review by YaLTeR — COMMENTED** [review]
> Hey, thought you might find this helpful. I ported this for my own use case but seeing as it was relatively simple, I thought I might as well send over a PR.

Hey! This is very helpful, thanks!

> I've tried to keep the project as like yours as possible, so I've kept the gradle setup... Even though this forced me to do a kinda ugly hack around the mods.toml file changes between Forge and Neoforge.

Extra thanks for this. It looks pretty much as I would've probably written it.

> It's worth noting, remapping for your use case would likely be fine but in the future, the two versions will likely need larger changes.

That's fine, if it breaks bad enough maybe I'll end up just dropping Forge, that seems to be the direction people are moving in, right?

File: src/main/java/yalter/mousetweaks/neoforge/MouseTweaksForge.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import net.neoforged.fml.ModLoadingContext;
+import net.neoforged.fml.loading.FMLEnvironment;
+import net.neoforged.fml.common.Mod;
+import yalter.mousetweaks.Constants;
+import yalter.mousetweaks.Logger;
+import yalter.mousetweaks.Main;
+import yalter.mousetweaks.MouseButton;
+
+@Mod(Constants.MOD_ID)
+public class MouseTweaksForge {
```
[the remark this answers is not in the document]
Comment (MichaelHillcox) [reply src/main/java/yalter/mousetweaks/neoforge/MouseTweaksForge.java]: This should be resolved now :+1:

**Comment by MichaelHillcox** [comment]
> That's fine, if it breaks bad enough maybe I'll end up just dropping Forge, that seems to be the direction people are moving in, right?

Yeah, kinda, although for as little as this mod reaches into the specific mod loaders, it looks like you have a good setup for it :D

**Comment by YaLTeR** [comment]
I tested it, seems to work. One thing though: NeoForge complains that

> Reference map 'mousetweaks.mixins.refmap.json' for mousetweaks.mixins.json could not be read. If this is a development environment you can ignore this message

And indeed that file is missing from the neoforge jar, but is present in the forge jar. Is that expected?

Also I guess not something we can fix here, but with both forge and neoforge jars in the mods folder, NeoForge tries to load the forge one and fails, and then doesn't try to load the neoforge jar.

**Comment by MichaelHillcox** [comment]
The refmap actually isn't needed on neoforge any more so it might be that we could just patch out that line using process resources, it's hacky but it does mean you wouldn't need to duplicate the file

**Comment by YaLTeR** [comment]
Ah then it's probably fine.

**Comment by YaLTeR** [comment]
Thanks again! Gonna push to CurseForge and modrinth today.

**Comment by YaLTeR** [comment]
How would you like to be credited in the Credits name list?

**Comment by MichaelHillcox** [comment]
If you want to add my name in credits, I typically go be ErrorMikey :+1:

**Comment by YaLTeR** [comment]
Pushed:
- https://www.curseforge.com/minecraft/mc-mods/mouse-tweaks/files/5156798
- https://modrinth.com/mod/mouse-tweaks/version/1.20.4-2.25-neoforge

Thanks again!

**Pull request merged**